ovgs.proto

```diff
@@ -1,175 +1,241 @@
+  rpc DeleteGroup(DeleteGroupRequest) returns (DeleteGroupResponse);
+  // GetGroup returns the domain-certs (keyed by id), serials,
+  // and user/role mappings for that group and all the children.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc GetGroup(GetGroupRequest) returns (GetGroupResponse);
+
+  // CreateUserRole assigns a role to a user in a named group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if user already exists in the group
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc CreateUserRole(AssignUserRoleRequest) returns (AssignUserRoleResponse);
+  // DeleteUserRole removes roles of a user in a named group.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
+  rpc DeleteUserRole(AssignUserRoleRequest) returns (AssignUserRoleResponse);
+  // GetUserRole returns the roles the user is assigned.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc GetUserRole(GetUserRoleRequest) returns (GetUserRoleResponse);
+
+  // CreateSerial assigns the serial to that group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if serial_number already exists in the group
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc CreateSerial(CreateSerialRequest) returns (CreateSerialResponse);
+  // DeleteSerial removes the serial from the group.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
+  rpc DeleteSerial(DeleteSerialRequest) returns (DeleteSerialResponse);
+  // GetSerial returns all the facts about the serial number.
+  // Errors will be returned:
+  // NOT_FOUND if the group doesn't exist
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc GetSerial(GetSerialRequest) returns (GetSerialResponse);
+
+  // CreateDomainCert creates the certificate in the group.
+  // Errors will be returned:
+  // ALREADY_EXISTS if cert already exists in the group
+  // PERMISSION_DENIED if the user doesn't have access to the group
+  rpc CreateDomainCert(CreateDomainCertRequest) returns (CreateDomainCertResponse);
+  // DeleteDomainCert deletes the cert from the database.
+  // Errors will be returned:
+  // NOT_FOUND if the cert doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
+  rpc DeleteDomainCert(DeleteDomainCertRequest) returns (DeleteDomainCertResponse);
+  // GetDomainCert reveals the details of the certificate.
+  // NOT_FOUND if the cert doesn't exist
+  // PERMISSION_DENIED if user doesn't have access to the group
+  rpc GetDomainCert(GetDomainCertRequest) returns (GetDomainCertResponse);
+
+  // GetOwnershipVoucher issues ownership voucher and returns TPM public key 
```

Comment:
returns TPM public key for the serial number -> returns TPM public key for the serial number (if it exists)?

Not all serial_numbers have a TPM public key associated with them